extensions/wikia/AchievementsII/services/AchRankingService.class.php

```diff
@@ -163,8 +163,12 @@ public function getRecentAwardedBadges($badgeLevel = null, $limit = null, $daySp
 		while(($row = $dbr->fetchObject($res)) && (count($badges) <= $limit)) {
 			$user = User::newFromId($row->user_id);
 
-			if( $user && AchAwardingService::canEarnBadges( $user ) ) {
-				$badges[] = array('badge' => new AchBadge($row->badge_type_id, $row->badge_lap, $row->badge_level), 'date' => $row->date);
+			if ( $user && AchAwardingService::canEarnBadges( $user ) ) {
+				$badges[] = [
+					'user' => $this->extractNeededUserData( $user ),
```

Comment:
FYI @sqreek, @jcellary, I'm adding back userData here, but only what's required for the template.